index.js

```diff
@@ -1,12 +1,35 @@
-var redis = require('redis');
+const redis = require('redis');
+const assert = require('assert');
+
+function assertBrokerOptions (brokerOptions) {
+  assert(brokerOptions, '"brokerOptions" is required to create a Redis client with sc-redis');
+  assert(brokerOptions.host, '"brokerOptions.host" is required to create a Redis client with sc-redis');
+  assert(brokerOptions.port, '"brokerOptions.port" is required to create a Redis client with sc-redis');
+}
+
+function throwMissingRedisClientError (clientName) {
+  throw new Error('Missing "' + clientName + '" option. Both "pubClient" and "subClient" must be specified if passing your own clients.');
+}
+
+module.exports.attach = function (broker, options) {
+  options = options || {}
+
+  const instanceId = broker.instanceId;
+  var subClient = options.subClient;
+  var pubClient = options.pubClient;
+
+  if (!subClient && !pubClient) {
```

Comment:
I think we should require that both `subClient` and `pubClient` options are passed if you choose to pass your own clients. Otherwise there is the potential for mixing and matching different Redis libraries, which could potentially cause strange behavior.